packages/toolpad-core/src/DashboardLayout/DashboardLayout.tsx

```diff
@@ -301,6 +310,30 @@ export interface DashboardLayoutProps {
    * The content of the dashboard.
    */
   children: React.ReactNode;
+  /**
+   * Branding options for the layout.
+   * @default null
+   */
+  branding?: AppProviderProps['branding'];
+  /**
+   * Navigation definition for the layout.
+   * @default []
+   */
+  navigation?: AppProviderProps['navigation'];
+  /**
+   * Active palette mode in theme.
+   */
+  paletteMode?: PaletteMode;
```

Comment:
I don't believe the state for `paletteMode` and `setPaletteMode` (we use a different naming convention for controlled props anyway) should be owned by `DashboardLayout`. It's owned by the theme and if the user wants to manipulate it `useColorScheme` should be used.

Reply:
Ok, it was just a first idea and i wasn't sure about the names for these 2 props either.
We can go with that, the only disadvantage I guess is that it will only work with MUI theming.

Reply:
I was looking a bit more into this and not sure about the approach to take:

1. User is using `DashboardLayout` inside an `AppProvider` with any of the 3 theming methods we support (https://mui.com/toolpad/core/react-app-provider/#theming) => theme switcher already works
2. User uses `DashboardLayout` without `AppProvider` around it => we can make things work automatically if they're using a CSS vars theme by simply using the methods from `useColorScheme`, otherwise if they're using a standard theme it can only be one theme unless they're managing the theme state themselves somehow, in which case we can't support it automatically?

About the shim, what scenario are we trying to solve with it? For internal components we have this context https://github.com/mui/mui-toolpad/blob/ad9bddb719a81cd113b73e82d304ae5ead1a0e8c/packages/toolpad-core/src/DashboardLayout/DashboardLayout.tsx#L81

Should I make an abstraction around the context as a hook that users can import called `useColorScheme`?

Reply:
Ok, we could make it a dump controlled property. But name it `colorScheme`, `onColorSchemeChange` to match naming of the cssvar theme, and also we always use `onXyzChange` for controlled prop `xyz`, not `setXyz`